File: www/cgi-bin/infinite.py
#!/usr/bin/env python3
"""
Infinite Loop CGI Script for WebServ - Python
This script tests timeout handling by running an infinite loop

WARNING: This script will run until killed by the server timeout!
"""

import sys
import hashlib
import time

# Print CGI header
print("Content-Type: text/html")
print()

print("<!DOCTYPE html>")
print("<html>")
print("<head><title>CGI Infinite Loop Test - Python</title></head>")
print("<body>")
print("<h1>Starting infinite loop...</h1>")

# Flush output
sys.stdout.flush()

# Infinite loop - server should kill this after timeout
counter = 0
while True:
    counter += 1
    # Do some busy work
    hash_obj = hashlib.md5(str(counter).encode())
    
    # Occasional output
    if counter % 1000000 == 0:
        print(f"<p>Iteration: {counter}</p>")
        sys.stdout.flush()
    
    # Safety limit (remove for true infinite loop testing)
    if counter > 100000000:
        print("<p>Safety limit reached</p>")
        break

print("</body>")
print("</html>")

File: www/cgi-bin/infinite.php
<?php
/**
 * Infinite Loop CGI Script for WebServ
 * This script tests timeout handling by running an infinite loop
 * 
 * WARNING: This script will run until killed by the server timeout!
 */

// Set max_execution_time to unlimited (for testing server-side timeout)
set_time_limit(0);

header("Content-Type: text/html");

echo "<!DOCTYPE html>\n";
echo "<html>\n";
echo "<head><title>CGI Infinite Loop Test</title></head>\n";
echo "<body>\n";
echo "<h1>Starting infinite loop...</h1>\n";

// Flush output buffer so client sees something
ob_flush();
flush();

// Infinite loop with sleep - server should kill this after timeout
$counter = 0;
while (true) {
    $counter++;
    sleep(1);  // Sleep 1 second per iteration
    echo "<p>Iteration: $counter seconds</p>\n";
    ob_flush();
    flush();
}

echo "</body>\n";
echo "</html>\n";
?>
